### DISCUSSION TITLE
Few improvments

### DISCUSSION THREAD

**Discussion by fdecourt** [docusealco/docuseal#19] [Ideas]
Hi,
It is great, but I think you need to add few stuff to make it useable : 

1. Sending an email must be mandatory. The creator of a document should not be able to sign without using an email, even if he is the only one signing. By copying the link, I am able to sign even if it is not me who should sign
2. We must be able to add the name and the email of the persons who will sign  (and show it below the signature)
3. When signing, below or above the signature, you should add the date and time of the signature, not modifiable by the user
4. with the signature you should also add an image saying something like : signed with CompanyName, or a logo of the company
5. _You should add an audit trail (date & time, signatory, email, IP address)._ > **Done** (great)

**Comment by lenny76** [comment]
Add multilanguage :)

**Comment by AlexBTurchyn** [comment]
@fdecourt thanks for those suggestions!

1. I think we should allow to sign documents in-person - so the party that is self-hosting with their admin account can enter into the signing form and hang their device to the signed to fill the form. But I agree that such 'in-person' signatures should also be verified via email - this will be added soon.

2. I still hesitate about this - adding first/last name will make it harder for admin users to add many signers in bulk. Usually firs/last name is collected in the process of filling the doc via the fields drawn on top of it - to collecting the same info twice seems redundant. But at the same time it would be nice to get the First/Last name to associate it with the signature. I'm open to discuss this suggestion more in depth since I'm still not sure which approach is better.

3. Automatic pkcs digital signatures that are added to the signed documents already serve this purpose - those signatures can be nread/verified at /settings/esign

4. Yeah, I've been thinking about adding this to the last page of the signed document.

5. Work in progress :)

**Reply by fdecourt** [reply]
Hi @AlexandrToorchyn ,
Thanks for your reply.

Point 1 seems absolutely necessary to me. An admin can get anyone to sign, in person or remotely. Maybe there should be two different processes, one remote, the other face-to-face (I'm a big user of remote electronic signature, but I've never done it face-to-face, so I'm not aware of the process!).

Points 2 and 3 are aimed at people / companies / public services who are not completely familiar with electronic signatures. The electronic file is protected, but, when you print it (and believe me, many still print documents), it's better to have a stamp indicating that it has been digitally signed (I have so many examples where this has been necessary, even if it's less and less frequent!).
The more information there is, the more reassuring it is for people (date, name and email adresse of signatory, name of signature provider, information that it has been signed electronically + signature)!
It's less pretty, but more reassuring!

I'd like to add a request: the possibility of adding an initials on each page.

**Comment by fdecourt** [comment]
More features : 

_6. the possibility of adding initials on each page._ > **Done** (should be able to upload an image to, like the point 7 below !

_7. The possiblity to upload an PNG image to be used as signature on the profil page. So, a user, if he must signs, will access directly to it's own signature picture_ > **Done** (with the possibility for admin user to save it in their profil (great !!!!)

**Comment by MrSquidOx** [comment]
Hi,

It would be helpful to be able to :

- automatically center the text in input text fields
- have the ability to rearrange the order of parts
- have the ability to choose if the Audit Log is sent or not

**Comment by omohokcoj** [comment]
>- We must be able to add the name and the email of the persons who will sign (and show it below the signature)
>- When signing, below or above the signature, you should add the date and time of the signature, not modifiable by the user
>- with the signature you should also add an image saying something like : signed with CompanyName, or a logo of the company

@fdecourt to address those 3 items we added a new field type - Stamp.
This is non interactive field type that is applied on the document page automatically once the document is signed.
Here is how this Stamp looks like on the page:
![image](https://github.com/docusealco/docuseal/assets/5418788/53e1d3f6-dd52-4283-8bce-97980d650ee2)
![image](https://github.com/docusealco/docuseal/assets/5418788/2e494d4a-963e-49b5-ae9a-fa1ec6ba5d0d)
